data/test/scenarios/events-test_filterable1.cfg

```diff
@@ -0,0 +1,198 @@
+# unit placed
+# prerecruit
+# recruit
+# exit_hex
+# enter_hex
+# moveto
+# attack
+# attacker hits
+# defender misses
+# attack end
+# attacker misses
+# defender hits
+# pre advance
+# advance
+# post advance
+{GENERIC_UNIT_TEST "events-test_filterable1" (
+    [event]
+        name=start
+# ensure attacker/defender hits/misses events are triggered
+        [modify_unit]
+            [filter]
+                name="Alice"
+            [/filter]
+            [effect]
+                apply_to=defense
+                replace=yes
+                [defense]
+                    flat=0
+                [/defense]
+            [/effect]
+        [/modify_unit]
+        [modify_unit]
+            [filter]
+                name="Bob"
+            [/filter]
+            [effect]
+                apply_to=max_experience
+                set=2
+            [/effect]
+            [effect]
+                apply_to=defense
+                replace=yes
+                [defense]
+                    castle=100
+                    frozen=100
+                    flat=100
+                [/defense]
+            [/effect]
+        [/modify_unit]
+    [/event]
+    [event]
+        name=unit placed
+        {VARIABLE unit_placed_flag 1}
+        {FAIL_IF_NOT unit_placed_flag 1}
+        {FAIL_IF_NOT prerecruit_flag $null}
+        {FAIL_IF_NOT recruit_flag $null}
+    [/event]
+    [event]
+        name=prerecruit
+        {VARIABLE prerecruit_flag 1}
+        {FAIL_IF_NOT unit_placed_flag 1}
+        {FAIL_IF_NOT prerecruit_flag 1}
+        {FAIL_IF_NOT recruit_flag $null}
+    [/event]
+    [event]
+        name=recruit
+        {VARIABLE recruit_flag 1}
+        {FAIL_IF_NOT unit_placed_flag 1}
+        {FAIL_IF_NOT prerecruit_flag 1}
+        {FAIL_IF_NOT recruit_flag 1}
+# ensure it's side 2's unit that dies
+        [modify_unit]
+            [filter]
+                x,y=$x1,$y1
+            [/filter]
+            [effect]
+                apply_to=defense
+                replace=yes
+                [defense]
+                    flat=0
+                [/defense]
+            [/effect]
+        [/modify_unit]
+        [do_command]
+            [move]
+                x=7,12
+                y=3,3
```

Comment:
i'm courious what this does, does it just teleport from 7,3 to 12,3 or does it actuall move the unit along a route from 7,3 to 12,3?

Reply:
If I run the test with `--showgui` then visibly it looks more like a teleport, however it's [documented](https://wiki.wesnoth.org/DirectActionsWML#.5Bdo_command.5D) that it should be treated like a regular move command from the user.  The test would also fail at any of the lines 184-186 if that weren't the case.

Reply:
Actuallyi just tested and it does in fact teleport, and by how do_command is implemented this also imples that a cheating player can teleport units around the map as he wants which is kinda bad.

So yes this will hopefully be fixed and then your code probably don't work anymore and you have  to specify an explicit route.

>  The test would also fail at any of the lines 184-186 if that weren't the case.

 enter_hex/exit_hey are also fired for 'teleport's, just not for the tiles between then.

Reply:
Either way should work, I'd think - what matters are the events are fired rather than how the unit got from hex A the hex B.

Reply:
> Either way should work, I'd think - what matters are the events are fired rather than how the unit got from hex A the hex B.

What im saying is that the expected outcome from that [do_command] is an error and that the unit will not move since, the unit cannot go in  one step  from (7,3) to (12,3). And as soon as we fix that your code will no longer work.

Reply:
Should be good now.